### DISCUSSION TITLE
How to Create Multiple Schedulers with Different Configurations in a Single Process using Quartz.NET and Microsoft's Default DI Container?

### DISCUSSION THREAD

**Discussion by ShayMusachanov-dev** [quartznet/quartznet#2134] [Q&A]
Hello everyone,

I'm trying to set up multiple Quartz.NET schedulers within the same process, each with its own unique configuration. I'm using Microsoft's default Dependency Injection container. My current approach involves calling AddQuartz multiple times, but I've noticed that the last call seems to override the previous ones.

Is there a way to create and manage multiple schedulers, each with its own unique name and configuration, within the same process without using clustering?

Any guidance or examples would be greatly appreciated. Thank you!

**Comment by hahn-kev** [comment]
I'm also interested in this. Some jobs I want to persist and make sure they execute. Other jobs are far less important and I want to trigger and store them in the RAM store as I don't want the performance hit of the sql store.

**Comment by lahma** [comment]
The new keyed registrations of MS DI with work for this. Probably would need something like `.AddQuartz(x => x.AddScheduler(name, s => { the configuration for specific scheduler} ))`

PRs proposing improvements are always welcome.

**Reply by hahn-kev** [reply]
would this make sense to do in Quartz 3 or 4? I'll hold of if it's 4 since it's not out yet.

**Reply by lahma** [reply]
Depends on how the new API shape would look like. If old usage still works and this new one is just extra API possibility it probably could go to 3.x too.
